File: android/src/main/java/com/jimmydaddy/imagemarker/base/RNImageSRC.kt
package com.jimmydaddy.imagemarker.base

import com.facebook.react.bridge.ReadableMap

data class RNImageSRC(val options: ReadableMap?) {

  var width: Int = 0
  var height: Int = 0
  var scale: Int = 1
  var uri: String = ""

  init {
    width = if (options?.hasKey("width") == true) options.getInt("width")!! else 0
    height = if (options?.hasKey("height") == true) options.getInt("height") else 0
    scale = if (options?.hasKey("scale") == true) options.getInt("scale") else 1
    uri = if (options?.hasKey("uri") == true) options.getString("uri").toString() else ""
  }
}

File: harmony/image_marker/src/main/ets/com.imagemarker.common/RNImageSRC.ts

export class RNImageSRC {
  width: number = 0;
  height: number = 0;
  scale: number = 1;
  uri: string = "";
  constructor(options: Map<string,object>) {
    this.width = options.has("width") ? Number(options.get("width")) : 0;
    this.height = options.has("height") ? Number(options.get("height")): 0;
    this.scale = options.has("scale") ?Number(options.get("scale")): 1;
    this.uri = options.has("uri") ? String(options.get("uri")): "";
  }
}